pyfa_tool/collection.py

```diff
@@ -0,0 +1,201 @@
+        of FaDatasets.
+
+        All FaDatasets are sorted along the validate dimension before merging.
+
+        Returns
+        -------
+        None.
+
+        Note
+        ------
+        For NWP and ensemble applications, the validate does not make an FA file
+        unique. So a combine on multiple dimensions is prefered for some of these
+        applications.
+
+        """
+        FaDatasets = self.FaDatasets
+
+        if len(FaDatasets) == 0:
+            sys.exit('No FaDatasets are provided.')
+        if len(FaDatasets) < 2:
+            sys.exit(f'Only one FaDatasets is provided: {FaDatasets[0]}')
+
+        # Sort Datasets by increasing validate
+        FaDatasets.sort(key=lambda x: x.get_validate(), reverse=False)
+        # TODO: For NWP there is need for an additional sorting on basetime!
+
+        # Prepare the attributes for merging
+        # 1. check if the vertical level defenitions is the same (A and B list)
+        comp_a = FaDatasets[0].ds.attrs['A_list']  # to compare with
+        comp_b = FaDatasets[0].ds.attrs['B_list']  # to copare with
+        for dataset in FaDatasets[1:]:
+            if not _check_lists_are_equal(comp_a, dataset.ds.attrs['A_list']):
+                sys.exit('Combining not possible since other defenition of levels is used (A_lists)')
+            if not _check_lists_are_equal(comp_b, dataset.ds.attrs['B_list']):
+                sys.exit('Combining not possible since other defenition of levels is used (B_lists)')
+
+            # Since they are equal, drop these attributes in all (but one) datasets
+            dataset._drop_attr('B_list')
+            dataset._drop_attr('A_list')
+
+        # get all attributes that are specific to one dataset
+        specific_comb_attributes = {'origins': [],
+                                    'filepaths': [],
+                                    }
+
+        # Make all Fadataset ready for merging on time coordinate
+        for dataset in FaDatasets:
+            specific_comb_attributes['origins'].append(dataset.ds.attrs['origin'])
+            dataset._drop_attr('origin')
+
+            specific_comb_attributes['filepaths'].append(dataset.ds.attrs['filepath'])
+            dataset._drop_attr('filepath')
+
+        ds = xr.concat(objs=[x.ds for x in FaDatasets],
+                       dim='validate',  # For NWP, 'basetime' must be added as dimension!
+                       data_vars='all',
+                       coords='all',
+                       compat='equals',
+                       positions=None,  # sort in advance !!
+                       fill_value=np.nan,
+                       join='outer',
+                       combine_attrs='no_conflicts'
+                       )
+
+        self.ds = ds
+        self._clean()
+        self.ds.attrs.update(specific_comb_attributes)
+
+    def _clean(self):
+        """Force a specific data format."""
+        self.ds = self.ds.transpose('y', 'x', 'level', 'validate', 'basedate')
+
```

Comment:
Future function to_netcdf similar to fa_dataset would be nice. However, as it is an xarray using xarrays to_netcdf should be sufficient. #48